src/Utils/Helpers.ts

```diff
@@ -230,6 +230,31 @@ export class Helpers {
     }, {});
   }
 
+  // parseDuration parses a duration string.
+  // a duration string is a sequence of decimal numbers,
+  // each with optional fraction and a unit suffix,
+  // such as "300ms" or "2h45m".
+  // valid time units are "s", "m", "h".
+  public static parseDuration(duration: string): number {
```

Comment:
All regexes below has polynomial complexity and possible vulnerable. It is not clear why `ms` package was not used as before.

Reply:
resolving thhis thread, it's not actual anymore.